risc0/zkvm/src/prove/mod.rs

```diff
@@ -176,196 +171,45 @@ where
     pub eval: Rc<E>,
 }
 
-/// TODO
+/// A Prover can execute a given [MemoryImage] and produce a [Receipt] that can
+/// be used to verify correct computation.
 pub trait Prover {
```

Comment:
:tada: I like the new `Prover` and the various `prove` wrappers, especially `prove_elf`, are great